internal/discharger/discharger.go

```diff
@@ -33,7 +32,15 @@ type MacaroonDischargerConfig struct {
 	ControllerUUID         string
 }
 
-func NewMacaroonDischarger(cfg MacaroonDischargerConfig, db *db.Database, ofgaClient *openfga.OFGAClient) (*MacaroonDischarger, error) {
+// UserMappingManager is an interface that defines methods for mapping local users to external users.
+type UserMappingManager interface {
+	// MapLocalUserToExternalUser maps a local user to an external user
+	// based on the user tag and the offer tag. It retrieves the model_uuid from the offer,
+	// and uses it to find the external user in the user mapping table.
+	MapLocalUserToExternalUser(ctx context.Context, userTag names.UserTag, offerTag names.ApplicationOfferTag) (*dbmodel.Identity, error)
```

Comment:
The map is from model-uuid + local-user -> external user. Nothing to do with application offer. If you want the manager to handle finding the model from the offer, that could work but if possible I would avoid to keep things a bit clearer.